pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/UnnecessaryVarargsArrayCreation.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+    xmlns="http://pmd.sourceforge.net/rule-tests"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+
+    <test-code>
+        <description>Unnecessary in asList</description>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>6,7</expected-linenumbers>
+        <code><![CDATA[
+            import java.util.Arrays;
+
+            public class Foo {
+                static {
+                    Arrays.asList("a" , "b");
+                    Arrays.asList(new String[] { "a", "b" });
+                    Arrays.asList(new Object[] { "a", "b" });
+                }
+            }
+            ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Necessary array creation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            import java.util.Arrays;
+
+            public class Foo {
+                static {
+                    Arrays.asList(new String[]{""}, new String[] { "a", "b" });
+                }
+            }
+            ]]></code>
+    </test-code>
+
+
+    <test-code>
+        <description>Confusing argument</description>
+        <expected-problems>2</expected-problems>
+        <expected-messages>
+            <message>Unnecessary explicit varargs array creation</message>
+            <message>Unclear if a varargs or non-varargs call is intended. Cast to Object or Object[], or pass varargs parameters separately to clarify intent.</message>
+        </expected-messages>
+        <code><![CDATA[
+            import java.util.Arrays;
+
+            public class Foo {
+                static {
+                    foo(new Object[]{ "a" }); // regular unnecessary
+                    foo(new String[]{ "a" }); // confusing bc String[] <: Object
```

Comment:
I was stumbling across this case:

https://github.com/spring-projects/spring-framework/blob/v5.3.13/spring-beans/src/test/java/org/springframework/beans/factory/config/MethodInvokingFactoryBeanTests.java#L258

But reading the message, I think, it's a reasonable suggestion. In the spring example, casting explicitly to `(Object)` should be ok.